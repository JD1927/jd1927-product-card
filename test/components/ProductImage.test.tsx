import React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/products';

describe('ProductImage', () => {
  test('Should show the component successfully with the custom image', () => {
    // Arrange
    const wrapper = renderer.create(
      <ProductImage image='https://image.jpg' />
    );
    // Act
    // Assert
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  test('Should show the component with the product image', () => {
    // Arrange
    const wrapper = renderer.create(
      <ProductCard product={product2}>
        {
          () => (<ProductImage />)
        }
      </ProductCard>
    );
    // Act
    // Assert
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
