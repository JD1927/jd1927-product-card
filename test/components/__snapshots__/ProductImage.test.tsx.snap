// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Should show the component successfully with the custom image 1`] = `
<img
  alt="Product"
  className="productImg undefined"
  src="https://image.jpg"
/>
`;

exports[`ProductImage Should show the component with the product image 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
